packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -283,125 +283,137 @@ describe('KeyringController', () => {
   });
 
   describe('importAccountWithStrategy', () => {
-    it('should import account with strategy privateKey', async () => {
-      await expect(
-        keyringController.importAccountWithStrategy(
-          AccountImportStrategy.privateKey,
-          [],
-        ),
-      ).rejects.toThrow('Cannot import an empty key.');
+    describe('using strategy privateKey', () => {
+      describe('when correct key is provided', () => {
+        it('should import account', async () => {
+          const address = '0x51253087e6f8358b5f10c0a94315d69db3357859';
+          const newKeyring = { accounts: [address], type: 'Simple Key Pair' };
+          const { keyringState, importedAccountAddress } =
+            await keyringController.importAccountWithStrategy(
+              AccountImportStrategy.privateKey,
+              [privateKey],
+            );
+          const modifiedState = {
+            ...initialState,
+            keyrings: [initialState.keyrings[0], newKeyring],
+          };
+          expect(keyringState).toStrictEqual(modifiedState);
+          expect(importedAccountAddress).toBe(address);
+        });
 
-      await expect(
-        keyringController.importAccountWithStrategy(
-          AccountImportStrategy.privateKey,
-          ['123'],
-        ),
-      ).rejects.toThrow(
-        'Expected private key to be an Uint8Array with length 32',
-      );
+        it('should not select imported account', async () => {
```

Comment:
Nit: Should we repeat this test for the JSON input as well? Or maybe bring it to the top-level of the `importAccountWithStrategy` block to show that it's not expected to differ between formats?

Reply:
Uhm, good point - I think it would make more sense to add it for the JSON input, rather than moving it to the top-level group because otherwise it would go along with:
* using strategy privateKey
* using strategy json
* using unrecognized strategy